### DISCUSSION TITLE
Enabling Inline Autofill in the browser extension

### DISCUSSION THREAD

**Discussion by fdisamuel** [dani-garcia/vaultwarden#4215] [Q&A] [answered]
Hi Everybody!

I read the new inline Autofill has been finally implemented into the browser extension, however I wasn't able to turn it on as the option is currently missing from the client logged into my self hosted vaultwarden instance. 

![image](https://github.com/dani-garcia/vaultwarden/assets/62856267/8ee80abc-e5cc-4be1-8d72-7128b79ab2d4)

In contrast, my cloud-connected browser extension shows the option:

![image](https://github.com/dani-garcia/vaultwarden/assets/62856267/9a31b16a-b798-4c1d-bd91-c7efb7b69f8d)

I suppose there is a version check on client side at least which prevents the function from showing up in the menus when connected to an older server version. 

Also [in the forum post](https://community.bitwarden.com/t/just-released-new-inline-auto-fill/61436) there is a mention for self-hosted installations 
> The feature will be available in self-hosted installations in the near future.

Does it mean that vaultwarden instances are not yet supported at all?

Thanks in advance if anyone can shed some light on the situation!

**Comment by stefan0xC** [comment, marked as answer]
We're currently considering providing a more flexible solution in regards to feature flags. See https://github.com/dani-garcia/vaultwarden/pull/4168 which would allow users to enable feature flags as they see fit. Once this is merged you'd need to enable both `autofill-v2` and `autofill-overlay` to enable this config option.

**Reply by fdisamuel** [reply]
Thank you so much for the clarifiaction! I will follow the pull request thread you linked, hopefully it gets merged soon :)

**Reply by tessus** [reply]
It's been approved, so it shouldn't take that long. Knock on wood.

**Reply by tippfehlr** [reply]
Just wanted to leave here that as of now this is in testing, but not in latest yet. Just use `ghcr.io/dani-garcia/vaultwarden:testing` for the image.
